src/algorithms/cache.rs

```diff
@@ -0,0 +1,291 @@
+use crate::{Correctness, Guess, Guesser, DICTIONARY, MAX_MASK_ENUM};
+use once_cell::sync::OnceCell;
+use std::borrow::Cow;
+use std::cell::Cell;
+use std::num::NonZeroU8;
+
+static INITIAL: OnceCell<Vec<(&'static str, f64, usize)>> = OnceCell::new();
+static PATTERNS: OnceCell<Vec<[Correctness; 5]>> = OnceCell::new();
+
+#[derive(Copy, Clone)]
+struct CacheValue(NonZeroU8);
+
+impl CacheValue {
+    fn new(val: u8) -> Self {
+        Self(NonZeroU8::new(val.wrapping_add(1)).unwrap())
+    }
+
+    fn get(&self) -> u8 {
+        self.0.get() - 1
+    }
+}
+
+const NUM_WORDS: usize = DICTIONARY.len();
+const CELL: Cell<Option<CacheValue>> = Cell::new(None);
+const ROW: [Cell<Option<CacheValue>>; NUM_WORDS] = [CELL; NUM_WORDS];
+thread_local! {
+    static COMPUTES: [[Cell<Option<CacheValue>>; NUM_WORDS]; NUM_WORDS] = [ROW; NUM_WORDS];
```

Comment:
This is about 170 MB, so it's not terribly surprising it doesn't fit nicely on the stack of any given thread. I think we may need to store this on the heap instead, and then make the thread local something like `once_cell::unsync::OnceCell<Box< [[Cell<Option<CacheValue>>; NUM_WORDS]; NUM_WORDS]>>`.

Reply:
This only occurs in dev builds, but I just had the idea to set the `opt-level` to 1 for dev builds. This resolves the issue, but I don't know if that is a suitable solution or not.

Reply:
I think allocating this on the heap is the right thing to do anyway. Stack memory is limited, and this is pretty large.